lib/linguist/file_blob.rb

```diff
@@ -26,6 +26,11 @@ def mode
       @mode ||= File.stat(@fullpath).mode.to_s(8)
     end
 
+    def symlink?
+      return @symlink if !@symlink.nil?
+      @symlink = (File.symlink?(@fullpath) rescue false)
+    end
```

Comment:
@kivikakk I just noticed this is reporting a bunch of warnings in the tests: https://travis-ci.org/github/linguist/jobs/323823193

```
/home/travis/build/github/linguist/lib/linguist/file_blob.rb:30: warning: instance variable @symlink not initialized
/home/travis/build/github/linguist/lib/linguist/file_blob.rb:30: warning: instance variable @symlink not initialized
/home/travis/build/github/linguist/lib/linguist/file_blob.rb:30: warning: instance variable @symlink not initialized
/home/travis/build/github/linguist/lib/linguist/file_blob.rb:30: warning: instance variable @symlink not initialized
```

They're not failures hence we missed this before, but I think it'd be best to clean these up.

Reply:
I've also just realised this PR didn't intro any tests to verify the symlink behaviour. Sorry I missed this before.

Reply:
No, my bad. I've cleaned up the uninitialised instance var messages in https://github.com/github/linguist/commit/16732843249d434969fbe5d21580a73fd4a73acf. Will add some tests in that PR.

Reply:
Also a test in https://github.com/github/linguist/pull/3948/commits/3eabf455a2f259e71669a4073896807e4d2ded94.